Forms

Chaseknis opened the pull request:
- Create a method called current_user in the ApplicationController
- Create forms to perform the following functions:
- [x] Creates a Post on behalf of the current_user.
- [x] Create a comment on behalf of the current_user.

- Add _comments_form.html.erb and _posts_form.html.erb
- Add comments_controller.rb and likes_controller.rb Update styles.css file
- Run linters for errors

Chaseknis committed:
Add _comments_form.html.erb and _posts_form.html.erb
Add comments_controller.rb and likes_controller.rb
Update styles.css file
Run linters for errors

FrederickMih commented on app/controllers/posts_controller.rb:
```diff
@@ -6,8 +6,38 @@ def index
 
+  def new
+    @post = Post.new
+    @current_user = ApplicationController.current_user
```
- [Optional] @Chaseknis, since the current_user method is defined in the ApplicationController, I think it should be accessible in all controllers hence you don't need to do this again.

FrederickMih commented on app/controllers/posts_controller.rb:
```diff
@@ -6,8 +6,38 @@ def index
+  end
+
+  def create
+    current_user = ApplicationController.current_user
```
- [Optional] kindly check the comments above :arrow_up:

FrederickMih commented on app/controllers/posts_controller.rb:
```diff
@@ -6,8 +6,38 @@ def index
+        title: post_params['title'],
+        text: post_params['text'],
+        likes_counter: 0,
+        comments_counter: 0
```
- [x] When I checked your posts table, I discovered that these columns were already defaulted to zero(0). It seems you are repeating yourself. Kindly remember the best practices for keeping our codebase DRY.

Whoistolu replied:
- Well resolved 👍🏽

FrederickMih commented on app/controllers/posts_controller.rb:
```diff
@@ -6,8 +6,38 @@ def index
+
+  def create
+    current_user = ApplicationController.current_user
+    @new_post =
+      Post.new(
+        author: current_user,
+        title: post_params['title'],
+        text: post_params['text'],
+        likes_counter: 0,
+        comments_counter: 0
+      )
```
- [Optional] You did a great job defining the `post_params` method for security purposes :+1: You may refactor your code to something of this sort: `@new_post = @current_user.posts.new(post_params)`

FrederickMih commented on app/controllers/posts_controller.rb:
```diff
@@ -6,8 +6,38 @@ def index
     @posts = @user.posts
   end
 
+  def new
+    @post = Post.new
+    @current_user = ApplicationController.current_user
+  end
+
+  def create
```
- [x] Based on the requirements of this project, you'll require a pair of routes for creating posts ie. posts#new, posts#create hence do not expose unnecessary routes in the routes.rb file.
- Hints: `resources :posts, only: [:new, :create] do` The rest consumed up our memory space.

Whoistolu replied:
- Well resolved 👍🏽

FrederickMih commented on app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,29 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @user = ApplicationController.current_user
+  end
+
+  def create
+    user = ApplicationController.current_user
+    post = Post.find(params[:post_id])
+    new_comment =
+      Comment.create(post: post, author: user, text: comment_params['text'])
```
- [Optional] As stated above, we can improve this :+1:

FrederickMih reviewed (changes requested):
## Hi :raising_hand:  @Chaseknis, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



<img src="https://media.giphy.com/media/nYYtQdLqwwlTe3ccKh/giphy.gif"  width ="150"><img>

## Highlights :green_circle: 

- PR is professional :+1: 
- Great job for using Rails partials :+1: 
- Descriptive README.MD file :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Chaseknis committed:
Update _posts_form.html.erb
Update comments_controller.rb, likes_controller.rb and posts_controller.rb
Run linters for errors

Whoistolu reviewed (approved):
### Hi @Chaseknis 

Great job fixing the changes recommended by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Chaseknis commented:
Thank you @Whoistolu

Whoistolu commented:
> Thank you @Whoistolu

You are welcome 🥇
